### PR TITLE
bitset rotation

### PR DESCRIPTION
**Pull request by micahscopes**
This is a very simple implementation of bitset rotation.  I don't think it's as fast as it could be, but it's more than fast enough for my purposes.

In [contemporary music theory](https://en.wikipedia.org/wiki/Post-tonal_music_theory), bitsets can be used to represent collections of pitches as subsets of a [cyclic order](https://en.wikipedia.org/wiki/Cyclic_order).

If `~` is used to mean equivalency, then in music theory, the following bitsets would in certain situations be thought of as equivalent: `1000 ~ 0100 ~ 0010 ~ 0001`.  In a 12-tone tuning system, you'd typically be working with length 12 bitsets, such as `101011010101`.

I didn't want to dig in too deeply, but if I were to do this "properly", maybe I'd build in rotation shifting in as a non-destructive modifier of BitSets, so that calling BitSet.get(idx) and BitSet.set(idx) would do an index shift, rather than copying onto a whole new BitSet.  But since I'm unfamiliar with the functions used in the rest of the library, I didn't have the motivation to get into it like that.  Plus, I'm going to be working with very small bitsets for now.

### CODE REVIEW

**Comment by mattkrick** [comment]
Thanks for the PR!
A circular shift would be a good addition. I think you're after something like this:
```
const circularShift = (bits, k) => (bits << k) | (bits >>> (32 - k));
```

A `forEach` with a modulo would become very expensive as bitsets get larger. Then for the final word, just do 1 modulo to figure out the additional right shift & perform that. I'm very open to the PR, but gotta keep the function living up to the name :smile:

**Comment by micahscopes** [comment]
Okay, I've made the change.  Should be "fast"-er now!

**Comment by micahscopes** [comment]
this needs more testing

**Comment by micahscopes** [comment]
What do you think about this version?  Is it still much slower to use forEach *without modulo* than it would be to do all the logic of bitshifting each word, etc. ?

**Comment by micahscopes** [comment]
FYI, I've still been working on word-by-word algorithm for this... it's quite a tricky challenge!  but i'm getting close.

**Comment by micahscopes** [comment]
Okay... I've implemented a word by word circular shift.  It was very hard, so my implementation is pretty verbose.

**Comment by mattkrick** [comment]
this is looking good! I'll try to dig in tonight, but it may take until the weekend. From an initial glance, I think we might be able to squeeze out a little more performance. I'll run some benchmarks before we merge.

**Comment by micahscopes** [comment]
any chance to look at this?

**Comment by mattkrick** [comment]
hey! thanks for the ping, i'll dig in now.

**Review by mattkrick — APPROVED** [review]

File: spec/BitsetSpec.js
Comment (micahscopes) [reply spec/BitsetSpec.js:267]: cool, I could write those

**Pull request merged**
